service/s3/s3manager/upload.go

```diff
@@ -547,14 +583,14 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker, firstPart []byte) (*Uploa
 
 	// Send part 1 to the workers
 	var num int64 = 1
-	ch <- chunk{buf: firstBuf, part: firstPart, num: num}
+	ch <- chunk{buf: firstBuf, part: firstPart, num: num, cleanup: cleanup}
 
 	// Read and queue the rest of the parts
 	for u.geterr() == nil && err == nil {
 		var reader io.ReadSeeker
 		var nextChunkLen int
 		var part []byte
-		reader, nextChunkLen, part, err = u.nextReader()
+		reader, nextChunkLen, part, cleanup, err = u.nextReader()
```

Comment:
I think it is possible for the `cleanup` not to be called if any of the error or `break` conditions are hit within this loop.   Would cause leaking of the cleanup's buffered bytes?